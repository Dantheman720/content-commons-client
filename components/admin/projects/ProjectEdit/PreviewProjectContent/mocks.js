export const getMocks = ( query, variables ) => (
  [
    {
      request: {
        query,
        variables: { ...variables }
      },
      result: {
        data: {
          project: {
            __typename: 'VideoProject',
            id: variables.id,
            projectType: 'LANGUAGE',
            descPublic: 'the project public description',
            team: {
              __typename: 'Team',
              id: 't81',
              name: 'the team name'
            },
            thumbnails: [
              {
                __typename: 'ImageFile',
                id: 'th11',
                createdAt: '2019-03-06T13:11:48.043Z',
                updatedAt: '2019-06-18T13:58:10.024Z',
                filename: 'image-1.jpg',
                filesize: 28371,
                filetype: 'image/jpeg',
                alt: 'the alt text',
                url: `2019/06/${variables.id}/image-1.jpg`,
                language: {
                  __typename: 'Language',
                  id: 'en38',
                  displayName: 'English',
                  languageCode: 'en',
                  locale: 'en-us',
                  nativeName: 'English',
                  textDirection: 'LTR'
                }
              }
            ],
            units: [
              {
                __typename: 'VideoUnit',
                id: 'un91',
                updatedAt: '2019-06-18T14:58:10.024Z',
                title: 'test project title',
                descPublic: 'the arabic description',
                language: {
                  __typename: 'Language',
                  id: 'ar22',
                  displayName: 'Arabic',
                  languageCode: 'ar',
                  locale: 'ar',
                  nativeName: 'العربية',
                  textDirection: 'RTL'
                },
                tags: [
                  {
                    __typename: 'Tag',
                    id: 'tag13',
                    translations: [
                      {
                        __typename: 'LanguageTranslation',
                        id: 'tr555',
                        name: 'الثقافة الأميركية',
                        language: {
                          __typename: 'Language',
                          id: 'ar22',
                          displayName: 'Arabic',
                          languageCode: 'ar',
                          locale: 'ar',
                          nativeName: 'العربية',
                          textDirection: 'RTL'
                        }
                      }
                    ]
                  }
                ],
                thumbnails: [
                  {
                    __typename: 'Thumbnail',
                    id: 'th11',
                    size: 'FULL',
                    image: {
                      __typename: 'ImageFile',
                      id: 'im28',
                      createdAt: '2019-03-06T13:11:48.043Z',
                      updatedAt: '2019-06-18T14:58:10.024Z',
                      filename: 'image-1.jpg',
                      filesize: 28371,
                      filetype: 'image/jpeg',
                      alt: 'the alt text',
                      url: `2019/06/${variables.id}/image-1.jpg`,
                      language: {
                        __typename: 'Language',
                        id: 'ar22',
                        displayName: 'Arabic',
                        languageCode: 'ar',
                        locale: 'ar',
                        nativeName: 'العربية',
                        textDirection: 'RTL'
                      }
                    }
                  }
                ],
                files: [
                  {
                    __typename: 'VideoFile',
                    id: 'f19',
                    createdAt: '2019-03-05T20:18:54.032Z',
                    updatedAt: '2019-06-19T17:38:37.502Z',
                    duration: 556,
                    filename: 'video-file-1.mp4',
                    filesize: 662595174,
                    filetype: 'video/mp4',
                    quality: 'WEB',
                    url: `2019/06/${variables.id}/video-file-1.mp4`,
                    videoBurnedInStatus: 'CLEAN',
                    dimensions: {
                      __typename: 'Dimensions',
                      id: 'd21',
                      height: '1080',
                      width: '1920'
                    },
                    language: {
                      __typename: 'Language',
                      id: 'ar22',
                      displayName: 'Arabic',
                      languageCode: 'ar',
                      locale: 'ar',
                      nativeName: 'العربية',
                      textDirection: 'RTL'
                    },
                    use: {
                      __typename: 'VideoUse',
                      id: 'us31',
                      name: 'Full Video'
                    },
                    stream: [
                      {
                        __typename: 'VideoStream',
                        id: 'st93',
                        site: 'YouTube',
                        url: 'https://www.youtube.com/watch?v=1evw4fRu3bo'
                      },
                      {
                        __typename: 'VideoStream',
                        id: 'st35',
                        site: 'Vimeo',
                        url: 'https://vimeo.com/340239507'
                      }
                    ]
                  }
                ]
              },
              {
                __typename: 'VideoUnit',
                id: 'un95',
                updatedAt: '2019-06-18T12:58:10.024Z',
                title: 'test project title',
                descPublic: 'the english description',
                language: {
                  __typename: 'Language',
                  id: 'en38',
                  displayName: 'English',
                  languageCode: 'en',
                  locale: 'en-us',
                  nativeName: 'English',
                  textDirection: 'LTR'
                },
                tags: [
                  {
                    __typename: 'Tag',
                    id: 'tag13',
                    translations: [
                      {
                        __typename: 'LanguageTranslation',
                        id: 'tr999',
                        name: 'american culture',
                        language: {
                          __typename: 'Language',
                          id: 'en38',
                          displayName: 'English',
                          languageCode: 'en',
                          locale: 'en-us',
                          nativeName: 'English',
                          textDirection: 'LTR'
                        }
                      },
                      {
                        __typename: 'LanguageTranslation',
                        id: 'tr019',
                        name: 'Culture américaine',
                        language: {
                          __typename: 'Language',
                          id: 'fr82',
                          displayName: 'French',
                          languageCode: 'fr',
                          locale: 'fr-fr',
                          nativeName: 'French',
                          textDirection: 'LTR'
                        }
                      }
                    ]
                  },
                  {
                    __typename: 'Tag',
                    id: 'tag14',
                    translations: [
                      {
                        __typename: 'LanguageTranslation',
                        id: 'tr888',
                        name: 'english learning',
                        language: {
                          __typename: 'Language',
                          id: 'en38',
                          displayName: 'English',
                          languageCode: 'en',
                          locale: 'en-us',
                          nativeName: 'English',
                          textDirection: 'LTR'
                        }
                      },
                      {
                        __typename: 'LanguageTranslation',
                        id: 'tr777',
                        name: 'Anglais langue étrangère',
                        language: {
                          __typename: 'Language',
                          id: 'fr82',
                          displayName: 'French',
                          languageCode: 'fr',
                          locale: 'fr-fr',
                          nativeName: 'French',
                          textDirection: 'LTR'
                        }
                      }
                    ]
                  }
                ],
                thumbnails: [
                  {
                    __typename: 'Thumbnail',
                    id: 'th11',
                    size: 'FULL',
                    image: {
                      __typename: 'ImageFile',
                      id: 'im28',
                      createdAt: '2019-03-06T13:11:48.043Z',
                      updatedAt: '2019-06-18T14:58:10.024Z',
                      filename: 'image-1.jpg',
                      filesize: 28371,
                      filetype: 'image/jpeg',
                      alt: 'the alt text',
                      url: `2019/06/${variables.id}/image-1.jpg`,
                      language: {
                        __typename: 'Language',
                        id: 'en38',
                        displayName: 'English',
                        languageCode: 'en',
                        locale: 'en-us',
                        nativeName: 'English',
                        textDirection: 'LTR'
                      }
                    }
                  }
                ],
                files: [
                  {
                    __typename: 'VideoFile',
                    id: 'f19',
                    createdAt: '2019-03-05T20:18:54.032Z',
                    updatedAt: '2019-06-19T17:38:37.502Z',
                    duration: 556,
                    filename: 'video-file-1.mp4',
                    filesize: 662595174,
                    filetype: 'video/mp4',
                    quality: 'WEB',
                    url: `2019/06/${variables.id}/video-file-1.mp4`,
                    videoBurnedInStatus: 'CLEAN',
                    dimensions: {
                      __typename: 'Dimensions',
                      id: 'd21',
                      height: '1080',
                      width: '1920'
                    },
                    language: {
                      __typename: 'Language',
                      id: 'en38',
                      displayName: 'English',
                      languageCode: 'en',
                      locale: 'en-us',
                      nativeName: 'English',
                      textDirection: 'LTR'
                    },
                    use: {
                      __typename: 'VideoUse',
                      id: 'us31',
                      name: 'Full Video'
                    },
                    stream: [
                      {
                        __typename: 'VideoStream',
                        id: 'st93',
                        site: 'YouTube',
                        url: 'https://www.youtube.com/watch?v=1evw4fRu3bo'
                      },
                      {
                        __typename: 'VideoStream',
                        id: 'st35',
                        site: 'Vimeo',
                        url: 'https://vimeo.com/340239507'
                      }
                    ]
                  }
                ]
              }
            ]
          }
        }
      }
    }
  ]
);
