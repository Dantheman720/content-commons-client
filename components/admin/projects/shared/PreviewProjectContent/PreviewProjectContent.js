/**
 *
 * PreviewProjectContent
 *
 */

import React from 'react';
import { object, string } from 'prop-types';
import { Dropdown, Embed } from 'semantic-ui-react';

import DownloadVideo from 'components/Video/DownloadVideo';
import DownloadSrt from 'components/Video/DownloadSrt';
import DownloadThumbnail from 'components/Video/DownloadThumbnail';
import DownloadOtherFiles from 'components/Video/DownloadOtherFiles';
import DownloadHelp from 'components/Video/DownloadHelp';

import ModalItem from 'components/modals/ModalItem/ModalItem';
import ModalContentMeta from 'components/modals/ModalContentMeta/ModalContentMeta';
import ModalDescription from 'components/modals/ModalDescription/ModalDescription';
import ModalPostMeta from 'components/modals/ModalPostMeta/ModalPostMeta';

import Notification from 'components/admin/projects/shared/Notification/Notification';
import PopupTrigger from 'components/popups/PopupTrigger';
import PopupTabbed from 'components/popups/PopupTabbed';

import downloadIcon from 'static/icons/icon_download.svg';
import { getYouTubeId } from 'lib/utils';

import './PreviewProjectContent.scss';


/* eslint-disable react/prefer-stateless-function */
class PreviewProjectContent extends React.PureComponent {
  constructor( props ) {
    super( props );

    const { data, projecttype } = this.props;

    this.state = {
      dropDownIsOpen: false,
      selectedLanguage: 'English',
      projectItems: this.getProjectItems( data, projecttype ),
      selectedItem: {},
      languages: this.getLanguages( data, projecttype )
    };
  }

  componentDidMount = () => {
    this.selectProjectItem();
  }

  getLanguages = ( obj, str ) => (
    obj[str].map( item => ( {
      key: item.language.language_code,
      value: item.language.display_name,
      text: item.language.display_name
    } ) )
  );

  getProjectItems = ( obj, str ) => (
    obj[str].reduce( ( acc, item ) => ( {
      ...acc,
      [item.language.display_name]: item
    } ), {} )
  );

  toggleArrow = () => {
    this.setState( prevState => ( {
      dropDownIsOpen: !prevState.dropDownIsOpen
    } ) );
  }

  selectLanguage = language => {
    this.setState(
      () => ( { selectedLanguage: language } ),
      this.selectProjectItem
    );
  }

  selectProjectItem = () => {
    const { projectItems, selectedLanguage } = this.state;
    this.setState( {
      selectedItem: projectItems[selectedLanguage]
    } );
  }

  handleChange = ( e, { value } ) => {
    this.toggleArrow();
    this.selectLanguage( value );
  }

  render() {
    const {
      projectType,
      projectData,
      updated,
      videos
    } = this.props.data;
    const { team } = projectData;

    const {
      dropDownIsOpen,
      selectedLanguage,
      selectedItem,
      languages
    } = this.state;

    if ( !selectedItem || !Object.keys( selectedItem ).length ) return null;

    const {
      title,
      thumbnail,
      language,
      desc,
      uploaded,
      source
    } = selectedItem;

    const youTubeUrl = source[0].streamUrl[0].url;
    const { burnedInCaptions } = source[0];

    const previewMsgStyles = {
      position: 'absolute',
      top: '0',
      left: '0',
      right: '0',
      // match Semantic UI border-radius
      borderTopLeftRadius: '0.28571429rem',
      borderTopRightRadius: '0.28571429rem',
      padding: '1em 1.5em',
      fontSize: '1em',
      backgroundColor: '#fdb81e'
    };

    return (
      <ModalItem
        className="project-preview"
        headline={ title }
        textDirection={ language.text_direction }
      >
        <Notification
          el="p"
          customStyles={ previewMsgStyles }
          msg={ `This is a preview of your ${projectType} project on Content Commons.` }
        />

        <div className="modal_options">
          <Dropdown
            className="modal_languages"
            value={ selectedLanguage }
            icon={ dropDownIsOpen ? 'chevron up' : 'chevron down' }
            options={ languages }
            onClick={ this.toggleArrow }
            onChange={ this.handleChange }
          />

          <div className="trigger-container">
            <PopupTrigger
              toolTip="Download video"
              icon={ { img: downloadIcon, dim: 18 } }
              position="right"
              show={ projectType === 'video' }
              content={ (
                <PopupTabbed
                  title="Download this video."
                  panes={ [
                    {
                      title: 'Video File',
                      component: (
                        <DownloadVideo
                          selectedLanguageUnit={ selectedItem }
                          instructions={ `Download the video and SRT files in ${selectedLanguage}.
                            This download option is best for uploading this video to web pages.` }
                          burnedInCaptions={ burnedInCaptions === 'true' }
                        />
                      )
                    },
                    {
                      title: 'SRT',
                      component: (
                        <DownloadSrt
                          instructions="Download SRTs"
                          units={ videos }
                        />
                      )
                    },
                    {
                      title: 'Thumbnail',
                      component: (
                        <DownloadThumbnail
                          instructions="Download Thumbnail(s)"
                          units={ videos }
                        />
                      )
                    },
                    {
                      title: 'Other',
                      component: (
                        <DownloadOtherFiles
                          instructions="Download Other File(s)"
                          units={ videos }
                        />
                      )
                    },
                    { title: 'Help', component: <DownloadHelp /> }
                  ] }
                />
              ) }
            />
          </div>
        </div>

        <div className="project-preview__content">
          { /* @todo getYouTubeId may not be necessary depending
            on how the YouTube URL is stored in data */ }
          { youTubeUrl
            && (
              <Embed
                id={ getYouTubeId( youTubeUrl ) }
                placeholder={ thumbnail }
                source="youtube"
              />
            ) }

          <ModalContentMeta type={ projectType } dateUpdated={ updated || uploaded } />

          <ModalDescription description={ desc } />
        </div>

        <ModalPostMeta source={ team } datePublished={ uploaded } />
      </ModalItem>
    );
  }
}

PreviewProjectContent.propTypes = {
  data: object.isRequired,
  projecttype: string
};

export default PreviewProjectContent;
