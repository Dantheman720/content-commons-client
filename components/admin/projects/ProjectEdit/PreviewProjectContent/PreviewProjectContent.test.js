import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import wait from 'waait';
import { MockedProvider } from 'react-apollo/test-utils';
import { Loader } from 'semantic-ui-react';
import PreviewProjectContent, { VIDEO_PROJECT_PREVIEW_QUERY } from './PreviewProjectContent';
import { getMocks } from './mocks';

jest.mock( 'lib/utils', () => ( {
  getPathToS3Bucket: jest.fn( () => {} ),
  getS3Url: jest.fn( assetPath => (
    `https://s3-url.com/${assetPath}`
  ) ),
  getStreamData: jest.fn( ( stream, site = 'youtube', field = 'url' ) => {
    const uri = stream.find( s => s.site.toLowerCase() === site );
    if ( uri && Object.keys( uri ).length > 0 ) {
      return uri[field];
    }
    return null;
  } ),
  getVimeoId: jest.fn( () => '340239507' ),
  getYouTubeId: jest.fn( () => '1evw4fRu3bo' ),
  contentRegExp: jest.fn( () => false )
} ) );

// jest.mock( 'static/icons/icon_download.svg', () => () => 'downloadIcon' );
// jest.mock( 'static/icons/icon_embed.svg', () => () => 'embedIcon' );
// jest.mock( 'static/icons/icon_share.svg', () => () => 'shareIcon' );

const props = { id: '123' };

const mocks = getMocks( VIDEO_PROJECT_PREVIEW_QUERY, props );

const vimeoMocks = [
  {
    ...mocks[0],
    result: {
      data: {
        project: {
          ...mocks[0].result.data.project,
          units: [
            {
              ...mocks[0].result.data.project.units[0],
              files: [
                {
                  ...mocks[0].result.data.project.units[0].files[0],
                  stream: [
                    {
                      __typename: 'VideoStream',
                      id: 'st35',
                      site: 'Vimeo',
                      url: 'https://vimeo.com/340239507'
                    }
                  ]
                }
              ]
            }
          ]
        }
      }
    }
  }
];

const noStreamsMocks = [
  {
    ...mocks[0],
    result: {
      data: {
        project: {
          ...mocks[0].result.data.project,
          units: [
            {
              ...mocks[0].result.data.project.units[0],
              files: [
                {
                  ...mocks[0].result.data.project.units[0].files[0],
                  stream: []
                }
              ]
            }
          ]
        }
      }
    }
  }
];

const noFilesMocks = [
  {
    ...mocks[0],
    result: {
      data: {
        project: {
          ...mocks[0].result.data.project,
          units: [
            {
              ...mocks[0].result.data.project.units[0],
              files: []
            }
          ]
        }
      }
    }
  }
];

const noUnitsMocks = [
  {
    ...mocks[0],
    result: {
      data: {
        project: {
          ...mocks[0].result.data.project,
          units: []
        }
      }
    }
  }
];

const Component = (
  <MockedProvider mocks={ mocks } addTypename>
    <PreviewProjectContent { ...props } />
  </MockedProvider>
);

describe( '<PreviewProjectContent />', () => {
  it( 'renders initial loading state without crashing', () => {
    const wrapper = mount( Component );
    const preview = wrapper.find( 'PreviewProjectContent' );
    const loader = (
      <Loader
        active
        inline="centered"
        style={ { marginBottom: '1em' } }
      />
    );
    const msg = <p>Loading the project preview...</p>;

    expect( preview.exists() ).toEqual( true );
    expect( preview.contains( loader ) ).toEqual( true );
    expect( preview.contains( msg ) ).toEqual( true );
  } );

  it( 'renders error message if error is thrown', async () => {
    const errorMocks = [
      {
        request: {
          query: VIDEO_PROJECT_PREVIEW_QUERY,
          variables: { id: '123' }
        },
        result: {
          errors: [{ message: 'There was an error.' }]
        }
      }
    ];
    const wrapper = mount(
      <MockedProvider mocks={ errorMocks } addTypename={ false }>
        <PreviewProjectContent { ...props } />
      </MockedProvider>
    );
    // wait for the data and !loading
    await wait( 0 );
    wrapper.update();

    const preview = wrapper.find( 'PreviewProjectContent' );
    const errorComponent = preview.find( 'ApolloError' );

    expect( errorComponent.exists() ).toEqual( true );
    expect( errorComponent.contains( 'There was an error.' ) )
      .toEqual( true );
  } );

  it( 'renders final state without crashing', async () => {
    const wrapper = mount( Component );
    await wait( 0 );
    wrapper.update();

    const preview = wrapper.find( 'PreviewProjectContent' );

    expect( toJSON( preview ) ).toMatchSnapshot();
  } );

  it( 'sets dropDownIsOpen and selectedLanguage in initial state', async () => {
    const wrapper = mount( Component );
    await wait( 0 );
    wrapper.update();

    const preview = wrapper.find( 'PreviewProjectContent' );
    const inst = preview.instance();
    const { dropDownIsOpen, selectedLanguage } = inst.state;
    const { language } = mocks[0].result.data.project.units[1];

    expect( dropDownIsOpen ).toEqual( false );
    expect( selectedLanguage ).toEqual( language.displayName );
  } );

  it( 'sets language in state when mounted', async () => {
    const wrapper = mount( Component );
    await wait( 0 );
    wrapper.update();

    const preview = wrapper.find( 'PreviewProjectContent' );
    const inst = preview.instance();
    const { selectedLanguage } = inst.state;
    const { language } = mocks[0].result.data.project.units[1];

    expect( selectedLanguage ).toEqual( language.displayName );
  } );

  it( 'calling getLanguages gets the unit language(s)', async () => {
    const wrapper = mount( Component );
    await wait( 0 );
    wrapper.update();

    const preview = wrapper.find( 'PreviewProjectContent' );
    const inst = preview.instance();
    const spy = jest.spyOn( inst, 'getLanguages' );
    const { units } = mocks[0].result.data.project;

    const languages = inst.getLanguages( units );

    expect( spy ).toHaveBeenCalled();
    expect( languages ).toEqual( [
      { key: 'ar', value: 'Arabic', text: 'Arabic' },
      { key: 'en', value: 'English', text: 'English' }
    ] );
  } );

  it( 'calling getProjectUnits gets the projectUnits', async () => {
    const wrapper = mount( Component );
    await wait( 0 );
    wrapper.update();

    const preview = wrapper.find( 'PreviewProjectContent' );
    const inst = preview.instance();
    const spy = jest.spyOn( inst, 'getProjectUnits' );
    const { units } = mocks[0].result.data.project;

    const projectUnits = inst.getProjectUnits( units );

    expect( spy ).toHaveBeenCalled();
    expect( typeof projectUnits ).toEqual( 'object' );
    expect( Object.keys( projectUnits ).length ).toEqual( units.length );
    units.forEach( unit => {
      const { language: { displayName } } = unit;
      expect( projectUnits[displayName] ).toEqual( unit );
    } );
  } );

  it( 'handleChange updates selectedLanguage and selectedItem in state', async () => {
    const wrapper = mount( Component );
    await wait( 0 );
    wrapper.update();

    const preview = wrapper.find( 'PreviewProjectContent' );
    const inst = preview.instance();
    const { units } = mocks[0].result.data.project;

    const handleChangeSpy = jest.spyOn( inst, 'handleChange' );
    const toggleArrowSpy = jest.spyOn( inst, 'toggleArrow' );
    const selectLanguageSpy = jest.spyOn( inst, 'selectLanguage' );

    const e = {};
    const selection = { value: 'French' };

    inst.handleChange( e, selection );

    const { selectedLanguage, selectedItem } = inst.state;

    expect( handleChangeSpy ).toHaveBeenCalledWith( e, selection );
    expect( toggleArrowSpy ).toHaveBeenCalled();
    expect( selectLanguageSpy ).toHaveBeenCalledWith( selection.value );

    expect( selectedLanguage ).toEqual( selection.value );
    expect( selectedItem ).toEqual(
      units.find( unit => unit.language.displayName === selectedLanguage )
    );
  } );

  it( 'toggleArrow updates dropDownIsOpen in state & updates dropdown icon', async () => {
    const wrapper = mount( Component );
    await wait( 0 );
    wrapper.update();

    const preview = () => wrapper.find( 'PreviewProjectContent' );
    const dropdown = () => preview().find( 'Dropdown.modal_languages' );
    const inst = preview().instance();
    const spy = jest.spyOn( inst, 'toggleArrow' );

    expect( inst.state.dropDownIsOpen ).toEqual( false );
    expect( dropdown().prop( 'icon' ) ).toEqual( 'chevron down' );

    inst.toggleArrow();
    expect( spy ).toHaveBeenCalledTimes( 1 );
    expect( inst.state.dropDownIsOpen ).toEqual( true );
    wrapper.update();
    expect( dropdown().prop( 'icon' ) ).toEqual( 'chevron up' );

    inst.toggleArrow();
    expect( spy ).toHaveBeenCalledTimes( 2 );
    expect( inst.state.dropDownIsOpen ).toEqual( false );
    wrapper.update();
    expect( dropdown().prop( 'icon' ) ).toEqual( 'chevron down' );
  } );

  it( 'selectLanguage updates selectedLanguage in state', async () => {
    const wrapper = mount( Component );
    await wait( 0 );
    wrapper.update();

    const preview = wrapper.find( 'PreviewProjectContent' );
    const inst = preview.instance();
    const selectedLanguage = () => inst.state.selectedLanguage;
    const spy = jest.spyOn( inst, 'selectLanguage' );
    const { language: initialLang } = mocks[0].result.data.project.units[1];
    const { language: newLang } = mocks[0].result.data.project.units[0];

    expect( selectedLanguage() ).toEqual( initialLang.displayName );

    inst.selectLanguage( newLang.displayName );
    expect( spy ).toHaveBeenCalled();
    expect( selectedLanguage() ).toEqual( newLang.displayName );

    inst.selectLanguage( initialLang.displayName );
    expect( spy ).toHaveBeenCalled();
    expect( selectedLanguage() ).toEqual( initialLang.displayName );
  } );

  it( 'getEnglishIndex returns the index for the English language unit', async () => {
    const wrapper = mount( Component );
    await wait( 0 );
    wrapper.update();

    const preview = wrapper.find( 'PreviewProjectContent' );
    const inst = preview.instance();
    const spy = jest.spyOn( inst, 'getEnglishIndex' );
    const { units } = mocks[0].result.data.project;
    const englishIndex = inst.getEnglishIndex( units );

    expect( spy ).toHaveBeenCalled();
    expect( englishIndex ).toEqual( units.length - 1 );
  } );

  it( 'getContentType returns the correct content type', async () => {
    const wrapper = mount( Component );
    await wait( 0 );
    wrapper.update();

    const preview = wrapper.find( 'PreviewProjectContent' );
    const inst = preview.instance();
    const spy = jest.spyOn( inst, 'getContentType' );
    const videoType = inst.getContentType( 'VideoProject' );
    const noType = inst.getContentType();

    expect( spy ).toHaveBeenCalled();
    expect( videoType ).toEqual( 'video' );
    expect( noType ).toEqual( '' );
  } );

  it( 'renders the share icon and PopupTrigger', async () => {
    const wrapper = mount( Component );
    await wait( 0 );
    wrapper.update();

    const preview = wrapper.find( 'PreviewProjectContent' );
    const sharePopup = preview.find( 'PopupTrigger[tooltip="Share project"]' );

    expect( sharePopup.exists() ).toEqual( true );
  } );

  it( 'renders the embed icon and PopupTrigger', async () => {
    const wrapper = mount( Component );
    await wait( 0 );
    wrapper.update();

    const preview = wrapper.find( 'PreviewProjectContent' );
    const embedPopup = preview.find( 'PopupTrigger[toolTip="Embed video"]' );

    expect( embedPopup.exists() ).toEqual( true );
  } );

  it( 'does not render the embed icon and PopupTrigger if there are no YouTube and Vimeo urls', async () => {
    const wrapper = mount(
      <MockedProvider mocks={ noStreamsMocks } addTypename>
        <PreviewProjectContent { ...props } />
      </MockedProvider>
    );
    await wait( 0 );
    wrapper.update();

    const preview = wrapper.find( 'PreviewProjectContent' );
    const embedPopup = preview.find( 'PopupTrigger[toolTip="Embed video"]' );

    expect( embedPopup.exists() ).toEqual( false );
  } );

  it( 'renders the download icon and PopupTrigger', async () => {
    const wrapper = mount( Component );
    await wait( 0 );
    wrapper.update();

    const preview = wrapper.find( 'PreviewProjectContent' );
    const downloadPopup = preview.find( 'PopupTrigger[toolTip="Download video"]' );

    expect( downloadPopup.exists() ).toEqual( true );
  } );

  it( 'renders an embedded YouTube video', async () => {
    const wrapper = mount( Component );
    await wait( 0 );
    wrapper.update();

    const preview = wrapper.find( 'PreviewProjectContent' );
    const embed = preview.find( 'Embed' );
    const assetPath = mocks[0].result.data.project.thumbnails[0].url;
    const s3Bucket = 'https://s3-url.com';

    expect( embed.exists() ).toEqual( true );
    expect( embed.prop( 'id' ) ).toEqual( '1evw4fRu3bo' );
    expect( embed.prop( 'source' ) ).toEqual( 'youtube' );
    expect( embed.prop( 'placeholder' ) )
      .toEqual( `${s3Bucket}/${assetPath}` );
  } );

  it( 'renders an embedded Vimeo video if there is no YouTube url', async () => {
    const wrapper = mount(
      <MockedProvider mocks={ vimeoMocks } addTypename>
        <PreviewProjectContent { ...props } />
      </MockedProvider>
    );
    await wait( 0 );
    wrapper.update();

    const preview = wrapper.find( 'PreviewProjectContent' );
    const embed = preview.find( 'Embed' );
    const assetPath = vimeoMocks[0].result.data.project.thumbnails[0].url;
    const s3Bucket = 'https://s3-url.com';

    expect( embed.exists() ).toEqual( true );
    expect( embed.prop( 'id' ) ).toEqual( '340239507' );
    expect( embed.prop( 'source' ) ).toEqual( 'vimeo' );
    expect( embed.prop( 'placeholder' ) )
      .toEqual( `${s3Bucket}/${assetPath}` );
  } );

  it( 'renders a project thumbnail if there are no YouTube or Vimeo urls', async () => {
    const wrapper = mount(
      <MockedProvider mocks={ noStreamsMocks } addTypename>
        <PreviewProjectContent { ...props } />
      </MockedProvider>
    );
    await wait( 0 );
    wrapper.update();

    const preview = wrapper.find( 'PreviewProjectContent' );
    const embed = preview.find( 'Embed' );
    const thumbnail = preview.find( 'figure.modal_thumbnail' );
    const img = thumbnail.find( 'img.overlay-image' );
    const assetPath = noStreamsMocks[0].result.data.project.thumbnails[0].url;
    const { alt } = noStreamsMocks[0].result.data.project.thumbnails[0];
    const s3Bucket = 'https://s3-url.com';

    expect( embed.exists() ).toEqual( false );
    expect( thumbnail.exists() ).toEqual( true );
    expect( img.prop( 'src' ) ).toEqual( `${s3Bucket}/${assetPath}` );
    expect( img.prop( 'alt' ) ).toEqual( alt );
  } );

  it( 'renders a "no files message" if there are no files in the selected unit', async () => {
    const wrapper = mount(
      <MockedProvider mocks={ noFilesMocks } addTypename>
        <PreviewProjectContent { ...props } />
      </MockedProvider>
    );
    await wait( 0 );
    wrapper.update();

    const preview = wrapper.find( 'PreviewProjectContent' );
    const noFilesMsg = 'This project unit does not have any files to preview.';

    expect( preview.contains( noFilesMsg ) ).toEqual( true );
  } );

  it( 'renders a "no units message" if there are no units in the project', async () => {
    const wrapper = mount(
      <MockedProvider mocks={ noUnitsMocks } addTypename>
        <PreviewProjectContent { ...props } />
      </MockedProvider>
    );
    await wait( 0 );
    wrapper.update();

    const preview = wrapper.find( 'PreviewProjectContent' );
    const noUnitsMsg = 'This project does not have any units to preview.';

    expect( preview.contains( noUnitsMsg ) ).toEqual( true );
  } );
} );
