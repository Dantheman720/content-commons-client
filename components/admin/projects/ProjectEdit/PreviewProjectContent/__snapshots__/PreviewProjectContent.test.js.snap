// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PreviewProjectContent /> renders error message if error is thrown 1`] = `"Error! GraphQL error: There was an error."`;

exports[`<PreviewProjectContent /> renders final state without crashing 1`] = `
<div
  className="modal LTR project-preview"
>
  <h1
    className="modal_headline"
  >
    Video Title (English)
  </h1>
  <p
    style={
      Object {
        "backgroundColor": "#fdb81e",
        "borderTopLeftRadius": "0.28571429rem",
        "borderTopRightRadius": "0.28571429rem",
        "fontSize": "1em",
        "left": "0",
        "padding": "1em 1.5em",
        "position": "absolute",
        "right": "0",
        "top": "0",
      }
    }
  >
    This is a preview of your video project on Content Commons.
  </p>
  <div
    className="modal_options"
  >
    <div
      aria-expanded={false}
      className="ui dropdown modal_languages"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="listbox"
      tabIndex={0}
    >
      <div
        aria-live="polite"
        className="text"
        role="alert"
      >
        English
      </div>
      <i
        aria-hidden="true"
        className="chevron down icon"
        onClick={[Function]}
      />
      <div
        className="menu transition"
      >
        <div
          aria-checked={true}
          aria-selected={true}
          className="active selected item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            English
          </span>
        </div>
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            French
          </span>
        </div>
      </div>
    </div>
    <div
      className="trigger-container"
    >
      <span
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <button
          className="ui button trigger"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          tooltip="Download video"
        >
          <img
            alt="Download video"
            height={18}
            src={Object {}}
            width={18}
          />
           
        </button>
      </span>
    </div>
  </div>
  <div
    className="project-preview__content"
  >
    <div
      className="ui embed"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="video play icon"
      />
      <img
        className="placeholder"
        src="https://website.com/filename.jpg"
      />
    </div>
    <section
      className="modal_section modal_section--metaContent"
    >
      <div
        className="modal_meta_wrapper"
      >
        <div
          className="modal_meta"
        >
          <span
            className="modal_meta_content modal_meta_content--filetype"
          >
            File Type: video
          </span>
          <span
            className="modal_meta_content modal_meta_content--date"
          >
            Updated: March 14, 2019
          </span>
        </div>
      </div>
    </section>
    <section
      className="modal_section modal_section--description"
    >
      <p
        className="modal_description_text"
      >
        the English description
      </p>
    </section>
  </div>
  <section
    className="modal_section modal_section--postMeta"
  >
    <span
      className="modal_postmeta_content"
    >
      Source:
       
      the team name
    </span>
    <span
      className="modal_postmeta_content"
    >
      Date Published: March 14, 2019
    </span>
  </section>
</div>
`;
